multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildApplicationDeployModelStep.java

```diff
@@ -88,6 +91,27 @@ private Staging modifyHealthCheckType(Staging staging) {
         return staging;
     }
 
+    private void buildBindingNames(List<RequiredDependency> requiredDependencies, CloudApplicationExtended modifiedApp,
+                                   ProcessContext context) {
+        Map<String, String> bindingNames = new HashMap<>();
+        int i = 0;
+        for (RequiredDependency requiredDependency : requiredDependencies) {
+            if (modifiedApp.getBindingParameters()
+                           .containsKey(requiredDependency.getName())
+                && modifiedApp.getBindingParameters()
+                              .get(requiredDependency.getName())
+                              .get(SupportedParameters.BINDING_NAME) != null) {
+                bindingNames.put(modifiedApp.getServices()
+                                            .get(i++),
+                                 modifiedApp.getBindingParameters()
+                                            .get(requiredDependency.getName())
+                                            .get(SupportedParameters.BINDING_NAME)
+                                            .toString());
+            }
+        }
```

Comment:
this is hard to read. if you have the binding name + binding params as a separate class you can remove this, then in DetermineApplicationServiceBindingActionsStep just set the current binding name in the context and use it during binding